androidApp/src/main/java/com/example/tasky/android/login/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.tasky.android.login.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.tasky.android.login.screen.LoginScreenEvent
+import com.example.tasky.android.login.screen.LoginScreenState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class LoginViewModel : ViewModel() {
+    private val _screenStateFlow = MutableStateFlow(LoginScreenState())
+    val screenStateFlow = _screenStateFlow.asStateFlow()
+
+    private val _isLoginSuccessFlow = MutableStateFlow(false)
+    val isLoginSuccessFlow = _isLoginSuccessFlow.asStateFlow()
```

Comment:
Could it not make sense to include this boolean property `isLoginSuccess` inside your `LoginScreenState` class?

Reply:
because this boolean property is used to trigger the navigation event, which would not pass to the LoginScreen, so I create a seperate flow for it.

Reply:
I see 🤔 

I would say, since this state is not apart of the `LoginScreen`, that this is instead the responsibility of your `MainActivity`. Let it decide whether to navigate to the `LoginScreen` or `AgendaScreen` based on the user’s logged in status. What do you think?